File: src/pocketcloud/cloud/network/packet/impl/normal/PlayerTextPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\normal;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\data\PacketData;
use pocketcloud\cloud\network\packet\impl\type\TextType;
use pocketcloud\cloud\player\CloudPlayerManager;

class PlayerTextPacket extends CloudPacket {

    public function __construct(
        private string $player = "",
        private string $message = "",
        private ?TextType $textType = null
    ) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->write($this->player);
        $packetData->write($this->message);
        $packetData->writeTextType($this->textType);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->player = $packetData->readString();
        $this->message = $packetData->readString();
        $this->textType = $packetData->readTextType();
    }

    public function getPlayer(): string {
        return $this->player;
    }

    public function getMessage(): string {
        return $this->message;
    }

    public function getTextType(): TextType {
        return $this->textType;
    }

    public function handle(ServerClient $client): void {
        if (($player = CloudPlayerManager::getInstance()->get($this->player)) !== null) $player->send($this->message, $this->textType);
    }

    public static function create(string $player, string $message, TextType $textType): self {
        return new self($player, $message, $textType);
    }
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/PlayerTextPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.network.packet.AuthenticatedPacket;
import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.CloudboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.network.packet.CloudPacket;
import de.pocketcloud.shared.network.packet.type.TextType;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class PlayerTextPacket extends CloudPacket implements CloudboundPacket, ClientboundPacket, AuthenticatedPacket {

    private String player;
    private String title;
    private String body;
    private TextType type;
    private int fadeIn = 20;
    private int stay = 20;
    private int fadeOut = 5;

    public PlayerTextPacket(String player, String message, TextType type) {
        this.player = player;
        this.title = "";
        this.body = message;
        this.type = type;
    }

    public PlayerTextPacket(String player, String title, String body, TextType type) {
        this.player = player;
        this.title = title;
        this.body = body;
        this.type = type;
    }

    public PlayerTextPacket(String player, String title, String body, TextType type, int fadeIn, int stay, int fadeOut) {
        this.player = player;
        this.title = title;
        this.body = body;
        this.type = type;
        this.fadeIn = fadeIn;
        this.stay = stay;
        this.fadeOut = fadeOut;
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(player, title, body, type, fadeIn, stay, fadeOut);
    }

    @Override
    public void decodePayload(IPacketData packetData) {
        this.player = packetData.readString();
        this.title = packetData.readString();
        this.body = packetData.readString();
        this.type = packetData.readEnum(TextType.class);
        this.fadeIn = packetData.readInt();
        this.stay = packetData.readInt();
        this.fadeOut = packetData.readInt();
    }

    public static PlayerTextPacket create(String player, String message, TextType type) {
        return new PlayerTextPacket(player, message, type);
    }

    public static PlayerTextPacket create(String player, String title, String body, TextType type) {
        return new PlayerTextPacket(player, title, body, type);
    }

    public static PlayerTextPacket create(String player, String title, String body, TextType type, int fadeIn, int stay, int fadeOut) {
        return new PlayerTextPacket(player, title, body, type, fadeIn, stay, fadeOut);
    }
}
